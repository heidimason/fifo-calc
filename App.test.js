import React from 'react'
import renderer from 'react-test-renderer'

import FifoCalculator from './App'

describe('<FifoCalculator />', () => {
	const tree = renderer.create(<FifoCalculator />).toJSON()

	it('renders correctly', () => {
		expect(tree).toMatchSnapshot()
	})

	it('has 2 children', () => {
		expect(tree.children.length).toBe(2)
	})
})
