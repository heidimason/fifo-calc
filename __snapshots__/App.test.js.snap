// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FifoCalculator /> renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        closing={false}
        gestureVelocityImpact={0.3}
        importantForAccessibility="auto"
        onClose={[Function]}
        onGestureBegin={[Function]}
        onGestureCanceled={[Function]}
        onOpen={[Function]}
        onTransitionStart={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        transitionSpec={
          Object {
            "close": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 0.01,
                "restSpeedThreshold": 0.01,
                "stiffness": 1000,
              },
            },
            "open": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 0.01,
                "restSpeedThreshold": 0.01,
                "stiffness": 1000,
              },
            },
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "flex": 1,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#fff",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                  "top": 0,
                  "width": 3,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#222",
                        "flex": 1,
                        "justifyContent": "space-around",
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "marginTop": 30,
                            },
                            Object {
                              "fontSize": 53,
                            },
                            Object {
                              "color": "#fff",
                              "marginHorizontal": 40,
                            },
                          ]
                        }
                      >
                        FIFO Profit Calculator
                      </Text>
                    </View>
                    <View>
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 40,
                            },
                            Object {
                              "color": "#fff",
                              "marginHorizontal": 40,
                            },
                            Object {
                              "marginBottom": 10,
                            },
                          ]
                        }
                      >
                        Purchase
                      </Text>
                      <TextInput
                        allowFontScaling={true}
                        keyboardType="numeric"
                        placeholder="Number of Shares"
                        rejectResponderTermination={true}
                        selectionColor="#222"
                        style={
                          Array [
                            Object {
                              "marginBottom": 15,
                            },
                            Object {
                              "borderRadius": 7,
                            },
                            Object {
                              "borderColor": "transparent",
                              "borderWidth": 0,
                            },
                            Array [
                              Object {
                                "fontSize": 40,
                              },
                              Object {
                                "backgroundColor": "#fff",
                                "color": "#aaa",
                                "height": 45,
                                "marginHorizontal": 40,
                                "textAlign": "center",
                              },
                            ],
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                      <TextInput
                        allowFontScaling={true}
                        keyboardType="numeric"
                        placeholder="Price per Share"
                        rejectResponderTermination={true}
                        selectionColor="#222"
                        style={
                          Array [
                            Object {
                              "marginBottom": 15,
                            },
                            Object {
                              "borderRadius": 7,
                            },
                            Object {
                              "borderColor": "transparent",
                              "borderWidth": 0,
                            },
                            Array [
                              Object {
                                "fontSize": 40,
                              },
                              Object {
                                "backgroundColor": "#fff",
                                "color": "#aaa",
                                "height": 45,
                                "marginHorizontal": 40,
                                "textAlign": "center",
                              },
                            ],
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                    <View
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#0099ff",
                          "borderRadius": 7,
                          "height": 45,
                          "justifyContent": "center",
                          "marginHorizontal": 40,
                          "opacity": 0.3,
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#fff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            Object {
                              "fontSize": 40,
                            },
                          ]
                        }
                      >
                        Submit
                      </Text>
                    </View>
                    <View
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignSelf": "flex-end",
                          "height": 45,
                          "opacity": 0.3,
                          "top": 18,
                          "transform": Array [
                            Object {
                              "rotate": "270deg",
                            },
                          ],
                          "zIndex": 1,
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "backgroundColor": "#228b22",
                              "letterSpacing": 3,
                              "paddingHorizontal": 3,
                              "textTransform": "uppercase",
                              "top": 3,
                            },
                            Object {
                              "color": "#fff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            Object {
                              "fontSize": 27,
                            },
                            Object {
                              "position": "absolute",
                            },
                          ]
                        }
                      >
                        History
                      </Text>
                    </View>
                    <View>
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 40,
                            },
                            Object {
                              "color": "#fff",
                              "marginHorizontal": 40,
                            },
                            Object {
                              "marginBottom": 10,
                            },
                          ]
                        }
                      >
                        Sale
                      </Text>
                      <TextInput
                        allowFontScaling={true}
                        keyboardType="numeric"
                        placeholder="Number of Shares"
                        rejectResponderTermination={true}
                        selectionColor="#222"
                        style={
                          Array [
                            Object {
                              "marginBottom": 15,
                            },
                            Object {
                              "borderRadius": 7,
                            },
                            Object {
                              "borderColor": "transparent",
                              "borderWidth": 0,
                            },
                            Array [
                              Object {
                                "fontSize": 40,
                              },
                              Object {
                                "backgroundColor": "#fff",
                                "color": "#aaa",
                                "height": 45,
                                "marginHorizontal": 40,
                                "textAlign": "center",
                              },
                            ],
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                      <TextInput
                        allowFontScaling={true}
                        keyboardType="numeric"
                        placeholder="Price per Share"
                        rejectResponderTermination={true}
                        selectionColor="#222"
                        style={
                          Array [
                            Object {
                              "marginBottom": 15,
                            },
                            Object {
                              "borderRadius": 7,
                            },
                            Object {
                              "borderColor": "transparent",
                              "borderWidth": 0,
                            },
                            Array [
                              Object {
                                "fontSize": 40,
                              },
                              Object {
                                "backgroundColor": "#fff",
                                "color": "#aaa",
                                "height": 45,
                                "marginHorizontal": 40,
                                "textAlign": "center",
                              },
                            ],
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                    <View
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#ffa500",
                          "borderRadius": 7,
                          "height": 45,
                          "justifyContent": "center",
                          "marginHorizontal": 40,
                          "opacity": 0.3,
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#fff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            Object {
                              "fontSize": 40,
                            },
                          ]
                        }
                      >
                        Calculate
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 40,
                            },
                            Object {
                              "letterSpacing": 1,
                              "marginBottom": 30,
                              "textTransform": "uppercase",
                            },
                            Object {
                              "color": "#fff",
                              "marginHorizontal": 40,
                            },
                          ]
                        }
                      >
                        Profit: 
                        0
                      </Text>
                    </View>
                    <View
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#ff0000",
                          "borderRadius": 7,
                          "height": 45,
                          "justifyContent": "center",
                          "marginBottom": 30,
                          "marginHorizontal": 40,
                          "opacity": 0.3,
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#fff",
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            Object {
                              "fontSize": 40,
                            },
                          ]
                        }
                      >
                        Reset All
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityElementsHidden={false}
      importantForAccessibility="auto"
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</RNCSafeAreaView>
`;
